core/mondoo-okta-security.mql.yaml

```diff
@@ -1,379 +1,1985 @@
+        The default session lifetime is two hours. A countdown timer appears to users when there are five minutes of session time remaining.
 
-        Read a helpful comparison here: https://developer.okta.com/docs/guides/oin-sso-overview/
-      audit: |
-        __Use `cnspec run`__
+        ### Okta recommends 
 
-        Run this query:
+        A session lifetime of two hours or less.
 
-          ```bash
-          cnspec run okta -c "okta.applications.all(_.signOnMode == "OPENID_CONNECT" ||  _.signOnMode == "SAML_2_0")" --organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        ### Security impact 
 
-        __Use the cnspec shell__
+        High
 
-        1. Launch the cnspec shell:
+        ### End-user impact 
 
-          ```bash
-          cnspec shell okta ---organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        Moderate
 
-        2. Run this query:
+        A countdown timer appears to users when there are five minutes of session time remaining.
+    variants: 
+      - uid: mondoo-okta-security-okta-enforce-session-lifetime-runtime
+  - uid: mondoo-okta-security-okta-enforce-session-lifetime-runtime
+    title: Enforce a limited session lifetime for all policies
+    impact: 80
+    filters: asset.platform == "okta-org"
+    props:
+      - uid: maxSessionIdleMinutes
+        title: Okta Security - SignOn maxSessionIdleMinutes
+        mql: return 120
+      - uid: maxSessionLifetimeMinutes
+        title: Okta Security - Max session lifetime minutes 
+        mql: return 1440     
+    mql: |
+      okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionIdleMinutes"] <= props.maxSessionIdleMinutes ))
+      okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionLifetimeMinutes"] <= props.maxSessionLifetimeMinutes ))
+    docs:
+      remediation: 
+        - id: console
+          desc: |
+            ### Set the session lifetime for a policy
+            1. In the Admin Console, go to **Security > Authentication**.
+            2. Click **Sign On**.
+            3. Click **Add Rule** or **Edit** to modify an existing policy rule.
+            4. Under **Session expires after**, set the session lifetime duration in minutes, hours, or days.
+            5. Click **Create Rule** or **Save Rule** once your changes have been made.
```

Comment:
```suggestion
            5. Select **Create Rule** or **Save Rule** once your changes have been made.
```

Reply:
@tas50 these are not our steps. they are from Okta's docs. They are copy/pasted.